docs/proposed/http-storage-node-protocol.rst

```diff
@@ -576,6 +576,136 @@ Just like ``GET /v1/mutable/:storage_index``.
 Advise the server the data read from the indicated share was corrupt.
 Just like the immutable version.
 
+Sample Interactions
+-------------------
+
+Immutable Data
+~~~~~~~~~~~~~~
+
+1. Create a bucket for storage index ``AAAAAAAAAAAAAAAA`` to hold two immutable shares, discovering that share ``1`` was already uploaded::
```

Comment:
Why would you ever create a bucket twice? Wouldn't you only ever do it once?

Reply:
I think because of failure recovery or maybe some other kind of uncoordinated multi-writer situation (but probably failure recovery).  For Foolscap, if you lose your TCP connection, you lose your reference to all of the objects that let you write to these shares and the only way to get them again is with this API (which Foolscap calls `allocate_bucket`).

I think it's still a reasonable way to do it in HTTP because you won't _know_ what the server might have received from a previous attempt.  Saying "I want to do X, Y, Z" and having the server say "You can skip X it's done" seems good.  The alternative is either forcing X to be re-uploaded needlessly or having the client do some kind of query before uploading "Do you have X, Y, and Z?" which just adds another round-trip.

Now ... where in this documentation does it makes sense to put that information...